ftw/linkchecker/per_site_configuration.py

```diff
@@ -14,18 +16,29 @@ def __init__(self, plone_site, configuration):
         self.upload_path = ''
         self.logger = logging.getLogger(LOGGER_NAME)
 
+        self.obj = None
         self._setup_plone(plone_site)
-        self.obj = plone_site
         self.configuration = self._get_config_for_portal()
         if self.configuration:
             self._set_upload_path()
 
     def _setup_plone(self, plone_site_obj):
-        setRequest(aq_parent(plone_site_obj).REQUEST)
+        app = self._find_app(plone_site_obj)
```

Comment:
I _think_ this is OK, but not quite sure what you're doing it for, and it seems to be going the long way round (especially since `plone.api` is available)

Reply:
The thing is that there is no request on the site objs. Therefore I'm placing one there. The request is necessary to use un-/restrictedTraverse. I can't do that on the site but have to do that on the app to make it work. Another option would be to pass around the app instead of few aq_parent rounds.

I chose this option to have less info within the object (which can be found anyways) and also less parameters/objs being passed around.